### PR TITLE
feat: add response header in debug view

### LINKED ISSUE #1680: feat: show http response header in debug view
# Feature request

## Please describe your feature

Currently only the response body is displayed, consider displaying the response header and response code at the same time.

image

### PR DESCRIPTION
**Pull request by stu01509**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**
Add response header tab in debug view.

image

image

**Related issues**

fix/resolve #1680

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

File: web/src/pages/Route/components/DebugViews/DebugDrawView.tsx [hunk 1]
```diff
@@ -51,6 +51,7 @@ const DebugDrawView: React.FC<RouteModule.DebugDrawProps> = (props) => {
   const [authForm] = Form.useForm();
   const [headerForm] = Form.useForm();
   const [responseCode, setResponseCode] = useState<string>();
```
Comment (stu01509) [inline web/src/pages/Route/components/DebugViews/DebugDrawView.tsx:53]: Hi @juzhiyuan 

I found the `setResponseCode` function,  actually deal with the response body, not response code,
so can I rename it?

Comment (juzhiyuan) [reply web/src/pages/Route/components/DebugViews/DebugDrawView.tsx:53]: If it is, yes you can.

File: web/src/pages/Route/components/DebugViews/DebugDrawView.tsx [hunk 2]
```diff
@@ -74,7 +74,8 @@ const DebugDrawView: React.FC<RouteModule.DebugDrawProps> = (props) => {
     formDataForm.setFieldsValue(DEFAULT_DEBUG_PARAM_FORM_DATA);
     headerForm.setFieldsValue(DEFAULT_DEBUG_PARAM_FORM_DATA);
     authForm.setFieldsValue(DEFAULT_DEBUG_AUTH_FORM_DATA);
-    setResponseCode(`${formatMessage({ id: 'page.route.debug.showResultAfterSendRequest' })}`);
+    setResponseBody(`${formatMessage({ id: 'page.route.debug.showResultAfterSendRequest' })}`);
```
[the remark this answers is not in the document]
Comment (stu01509) [reply web/src/pages/Route/components/DebugViews/DebugDrawView.tsx:77]: Update the commit :)

**Comment by stu01509** [comment]
Hi @nic-chen 

Could you guide me how to solve **Backend E2E** test error when you have time?

File: web/src/pages/Route/components/DebugViews/DebugDrawView.tsx [hunk 3]
```diff
@@ -73,7 +74,8 @@ const DebugDrawView: React.FC<RouteModule.DebugDrawProps> = (props) => {
...
```
Comment (juzhiyuan) [inline web/src/pages/Route/components/DebugViews/DebugDrawView.tsx:77]:
```suggestion
    setResponseBody(formatMessage({ id: 'page.route.debug.showResultAfterSendRequest' }));
```

File: web/src/pages/Route/components/DebugViews/DebugDrawView.tsx [hunk 4]
```diff
@@ -73,7 +74,8 @@ const DebugDrawView: React.FC<RouteModule.DebugDrawProps> = (props) => {
...
+    setResponseHeader(`${formatMessage({ id: 'page.route.debug.showResultAfterSendRequest' })}`);
```
Comment (juzhiyuan) [inline web/src/pages/Route/components/DebugViews/DebugDrawView.tsx:78]:
```suggestion
    setResponseHeader(formatMessage({ id: 'page.route.debug.showResultAfterSendRequest' }));
```

**Comment by nic-chen** [comment]
> Hi @nic-chen 
> 
> 
> 
> Could you guide me how to solve **Backend E2E** test error when you have time?

hi @stu01509 

Just add header content to the assertion.

**Comment by stu01509** [comment]
> Just add header content to the assertion.

OK, I will do add it :)

**Comment by liuxiran** [comment]
Hi @stu01509 , thanks for your nice work.
since you added a new tab for the response results area, you have to modify the fe e2e test cases at the same time to fix the fe e2e test failure

image

Try to add `cy.get('#rc-tabs-1-tab-response').should('be.visible').click();` after each `cy.wait('@DebugAPI');` in the https://github.com/apache/apisix-dashboard/blob/master/web/cypress/integration/route/online-debug.spec.js

Or just put the `Response`(Response Body would be better) before `Response Header` is also a better choice, for most users will pay attention to the response body first.

**Comment by juzhiyuan** [comment]
good good

**Review by liuxiran — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Review by Jaycean — APPROVED** [review]
Great changes.

**Review by LiteSun — APPROVED** [review]

**Comment by liuxiran** [comment]
merged, thanks for your contribution~!

**Pull request merged**